File: baekjoon/Silver/9012. 괄호/괄호.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        
        int T = Integer.parseInt(br.readLine());
        for (int i=0; i<T; i++) {
            String str = br.readLine();
            Stack<Character> s = new Stack<>();
            boolean isRight = true;
            for (char c : str.toCharArray()) {
                if (c == '(')
                    s.push('(');
                else {
                    if (s.isEmpty()) {
                        bw.write("NO\n");
                        isRight = false;
                        break;
                    } else {
                        s.pop();
                    }   
                }
            }
            if (!s.isEmpty()) {
                bw.write("NO\n");
            } else if (isRight) {
                bw.write("YES\n");
            }
        }
        bw.close();
    }
}

File: baekjoon/Silver/9012. 괄호/괄호.py
"""
[Stack]
1. '('는 append / ')'는 pop
2. ')'일 때 stack 비어있거나, 마지막에 잔여 값이 있다면 NO
"""

import sys
input = sys.stdin.readline

T = int(input())
for _ in range(T):
    ps = input().strip()
    s = []
    VPS = True
    for i in range(len(ps)):
        if ps[i] == '(':
            s.append(ps[i])
        else:
            if s:
                s.pop()
            else:
                VPS = False
                break
    if VPS and not s:
        print("YES")
    else:
        print("NO")